Indicate that p-values are computed two-tailed

axrhart opened the pull request:
Hi,

we recommend MOTE to our undergraduates and saw that some of them use MOTE to conveniently compute p-values from sample parameters, but expect the p-value to be one-tailed (so they doubled the result). Since they are not experienced enough to check the code, I thought it might be a good idea if the docs indicate how p is computed.

I know, this is a only a minor addition and two-tailed results are standard in many statistics packages, but I hope you consider including it anyway.

Best
--- Alex

axrhart committed:
Extended t docs

axrhart committed:
Fixed gsl installation candidate for older linux

axrhart committed:
Added gsl wrapper package

axrhart commented:
I also tried fo fix the GSL error:

`libgsl-dev` is [not available in the official repos for linux versions before 16.04](https://packages.ubuntu.com/search?keywords=libgsl-dev). So I do not know how [this](https://travis-ci.org/doomlab/MOTE/builds/508172020?utm_source=github_status&utm_medium=notification) build could work at all. 

Yet, replacing `libgsl-dev` with its older counterpart does not fix the problem, as the new gsl R-package from CRAN (which MBESS asked for) seems to depend on some function, that is [only available in newer GSL versions](https://packages.ubuntu.com/search?keywords=libgsl2). 

Maybe you should consider updating to xenial or bionic. I tried to do this via travis.yml, but this simple change is not compatible with the repos that are set in travis.

doomlab commented:
> Hi,
> 
> we recommend MOTE to our undergraduates and saw that some of them use MOTE to conveniently compute p-values from sample parameters, but expect the p-value to be one-tailed (so they doubled the result). Since they are not experienced enough to check the code, I thought it might be a good idea if the docs indicate how p is computed.
> 
> I know, this is a only a minor addition and two-tailed results are standard in many statistics packages, but I hope you consider including it anyway.
> 
> Best
> --- Alex

Excellent idea - since your students have been using it - where do you think that might be best to put it? In the help page? (Or are you talking about the shiny app?)
@axrhart

doomlab commented:
> I also tried fo fix the GSL error:
> 
> `libgsl-dev` is [not available in the official repos for linux versions before 16.04](https://packages.ubuntu.com/search?keywords=libgsl-dev). So I do not know how [this](https://travis-ci.org/doomlab/MOTE/builds/508172020?utm_source=github_status&utm_medium=notification) build could work at all.
> 
> Yet, replacing `libgsl-dev` with its older counterpart does not fix the problem, as the new gsl R-package from CRAN (which MBESS asked for) seems to depend on some function, that is [only available in newer GSL versions](https://packages.ubuntu.com/search?keywords=libgsl2).
> 
> Maybe you should consider updating to xenial or bionic. I tried to do this via travis.yml, but this simple change is not compatible with the repos that are set in travis.

Good points - I have been struggling with the `gsl` issues with `MBESS` dependency. Right now, I'm going to leave it out and try the path I had been working (taking `MBESS` out completely, which makes the package smaller for something else we are doing). If that is not possible, I'll work on upgrading the Ubuntu options. Thanks for your help!

axrhart commented:
Thank you for considering the addition @doomlab.

> Excellent idea - since your students have been using it - where do you think that might be best to put it? In the help page? (Or are you talking about the shiny app?)

I thought about adding it to the help pages, since we teach them to use the help when they are stuck or don't know how to work with a function. To save you some time, I've already added the changes I intended to my fork of MOTE. You can see the changes to the files in [this commit](https://github.com/doomlab/MOTE/pull/6/commits/bf0085492376a31eb47ed2bea9eb64d8c2fcd9f7) (which is basically just adding '(two-tailed)' to 'p-value' in the results section :D). If you accept this pull request the changes should be automatically incorporated into your repository.

Please note that I also changed two parameters in the `travis.yml`. You may want to revert them back to your original parameters. But I don't expect it to run in travis either way (see the [travis error here](https://travis-ci.org/doomlab/MOTE/builds/527803831?utm_source=github_status&utm_medium=notification) with only my basic commit to the help pages).

The additions might also be useful in the code section of the shiny app. The calculations in the app, however, have never really worked for me (and also do not work right now). Nothing happens when I press calculate regardless of how many fields are filled with values, so I don't know if there are any references to the p-value in the summary. Maybe this is specific to my setup (Win 10 with Firefox, no Addons), but my own apps on shinyapps.io work fine for me.

> Good points - I have been struggling with the gsl issues with MBESS dependency. Right now, I'm going to leave it out and try the path I had been working (taking MBESS out completely, which makes the package smaller for something else we are doing). If that is not possible, I'll work on upgrading the Ubuntu options.

I see. I think the automated testing should work without MBESS, but it would be sad if you have to sacrifice some functionality of the package by dropping it. The linux package dependencies are a pain if you don't expect them (e.g. because you work on a windows machine all day, like me). Lately, I had some similar issues while setting up some RStudio Server containers. Let me know if I can assist you with that issue.

doomlab commented:
> I thought about adding it to the help pages, since we teach them to use the help when they are stuck or don't know how to work with a function. To save you some time, I've already added the changes I intended to my fork of MOTE. You can see the changes to the files in [this commit](https://github.com/doomlab/MOTE/pull/6/commits/bf0085492376a31eb47ed2bea9eb64d8c2fcd9f7) (which is basically just adding '(two-tailed)' to 'p-value' in the results section :D). If you accept this pull request the changes should be automatically incorporated into your repository.

I'll check it out soon! Things are nuts around here, but should be able to get to it in a week or so. To update on CRAN can be a little bit slower, so you can use:

`devtools::install_github("doomlab/MOTE")` to get the newest version until I do the CRAN update. 

> The additions might also be useful in the code section of the shiny app. The calculations in the app, however, have never really worked for me (and also do not work right now). Nothing happens when I press calculate regardless of how many fields are filled with values, so I don't know if there are any references to the p-value in the summary. Maybe this is specific to my setup (Win 10 with Firefox, no Addons), but my own apps on shinyapps.io work fine for me.
> 

Question here: are you entering a number for alpha?
<img width="171" alt="image" src="https://user-images.githubusercontent.com/27783894/57184742-d127c780-6e8d-11e9-9959-fa7dac7faa93.png">
The grayed out .05 is just a suggestion...you do have to enter a number there for it to calculate. We suppressed the error output for other reasons which is why you might not see that it's waiting for you to enter something. 

<img width="1152" alt="image" src="https://user-images.githubusercontent.com/27783894/57184749-f87e9480-6e8d-11e9-9a9a-51f861ca1c1c.png">
 
> I see. I think the automated testing should work without MBESS, but it would be sad if you have to sacrifice some functionality of the package by dropping it. The linux package dependencies are a pain if you don't expect them (e.g. because you work on a windows machine all day, like me). Lately, I had some similar issues while setting up some RStudio Server containers. Let me know if I can assist you with that issue.

We wouldn't be dropping any functionality - mostly telling it to pull the functions we need from MBESS without the added bulk of all the things that MBESS does that we aren't using. 

Appreciate your comments and thoughts.

axrhart commented:
> I'll check it out soon! Things are nuts around here, but should be able to get to it in a week or so. To update on CRAN can be a little bit slower, so you can use:
> devtools::install_github("doomlab/MOTE") to get the newest version until I do the CRAN update.

Thank you very much! I'll update our students' working environments as soon as the new version is available. :)

> Question` here: are you entering a number for alpha?

Yes, actually, it seems I provided too much information. The output of summary for the single sample t  vanishes as soon as i provide SD **and** SE. Or if I provide only SD/df, SE/n or SE/df. So basically, entering more valid numbers made it suppress the summary output.

I'm sorry, I should have discovered my error by watching the help video properly. Having said that, is this the intended behavior or is it a side effect of suppressing the other errors? 

> We wouldn't be dropping any functionality - mostly telling it to pull the functions we need from MBESS without the added bulk of all the things that MBESS does that we aren't using.

If I understand you correctly, you basically are going to import a subset of the package, like it can be done in other languages? That's exciting! I did not know that R was capable of doing that. But how can this be done without installing the whole package in a local library? Right now I can only think of the options to copy the code into MOTE or pulling it on the fly from some remote server any time a function is called.

doomlab committed:
Merge branch 'master' into master
